src/controllers/dynakube/connectioninfo/reconciler.go

```diff
@@ -56,121 +55,80 @@ func (r *Reconciler) reconcileOneAgentConnectionInfo() error {
+		log.Info(dynatracev1beta1.GetCacheValidMessage(
+			"activegate connection info update",
+			r.dynakube.Status.ActiveGate.ConnectionInfoStatus.LastRequest,
+			r.dynakube.FeatureApiRequestThreshold()))
+		return nil
 	}
 
-	err = r.createOneAgentTenantConnectionInfoConfigMap(r.dynakube.OneAgentConnectionInfoConfigMapName(), connectionInfo)
+	connectionInfo, err := r.dtc.GetActiveGateConnectionInfo()
 	if err != nil {
+		log.Info("failed to get activegate connection info")
 		return err
 	}
 
-	return nil
-}
+	r.updateDynakubeActiveGateStatus(connectionInfo)
 
-func (r *Reconciler) createActiveGateTenantConnectionInfoConfigMap(secretName string, connectionInfo dtclient.ConnectionInfo) error {
-	configMapData := extractPublicData(connectionInfo, "")
-	configMap, err := kubeobjects.CreateConfigMap(r.scheme, r.dynakube,
-		kubeobjects.NewConfigMapNameModifier(secretName),
-		kubeobjects.NewConfigMapNamespaceModifier(r.dynakube.Namespace),
-		kubeobjects.NewConfigMapDataModifier(configMapData))
-	if err != nil {
-		return errors.WithStack(err)
-	}
-
-	query := kubeobjects.NewConfigMapQuery(r.context, r.client, r.apiReader, log)
-	err = query.CreateOrUpdate(*configMap)
+	err = r.createTenantTokenSecret(r.dynakube.ActivegateTenantSecret(), connectionInfo.ConnectionInfo)
 	if err != nil {
-		log.Info("could not create or update configMap for connection info", "name", configMap.Name)
 		return err
 	}
+
+	log.Info("activegate connection info updated")
+	r.dynakube.Status.ActiveGate.ConnectionInfoStatus.LastRequest = metav1.Now()
 	return nil
 }
 
-func (r *Reconciler) createOneAgentTenantConnectionInfoConfigMap(secretName string, connectionInfo dtclient.OneAgentConnectionInfo) error {
-	communicationHosts, err := encodeCommunicationHosts(connectionInfo)
-	if err != nil {
-		return err
-	}
-	configMapData := extractPublicData(connectionInfo.ConnectionInfo, communicationHosts)
-	configMap, err := kubeobjects.CreateConfigMap(r.scheme, r.dynakube,
-		kubeobjects.NewConfigMapNameModifier(secretName),
-		kubeobjects.NewConfigMapNamespaceModifier(r.dynakube.Namespace),
-		kubeobjects.NewConfigMapDataModifier(configMapData))
-	if err != nil {
-		return errors.WithStack(err)
-	}
-
-	query := kubeobjects.NewConfigMapQuery(r.context, r.client, r.apiReader, log)
-	err = query.CreateOrUpdate(*configMap)
-	if err != nil {
-		log.Info("could not create or update configMap for connection info", "name", configMap.Name)
-		return err
-	}
-	return nil
+func (r *Reconciler) updateDynakubeActiveGateStatus(connectionInfo dtclient.ActiveGateConnectionInfo) {
+	r.dynakube.Status.ActiveGate.ConnectionInfoStatus.TenantUUID = connectionInfo.TenantUUID
+	r.dynakube.Status.ActiveGate.ConnectionInfoStatus.TenantToken = connectionInfo.TenantToken
+	r.dynakube.Status.ActiveGate.ConnectionInfoStatus.Endpoints = connectionInfo.Endpoints
 }
 
 func (r *Reconciler) createTenantTokenSecret(secretName string, connectionInfo dtclient.ConnectionInfo) error {
```

Comment:
you moved out the config map creation to their respective reconcilers (to the activegate/oneagent reconcilers)
in order to make the connectioninfo reconciler only care about updating the status, but `createTenantTokenSecret` remained. Why ?